src/components/Footer/Footer.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useCallback, useContext } from 'react';
+import classNames from 'classnames';
+
+import { deleteTodo } from '../../api/todos';
+import { AppContext } from '../AppContext';
+import { TodosFilter } from '../TodosFilter';
+
+export const Footer: React.FC = () => {
+  const {
+    allTodos,
+    setAllTodos,
+    activeTodos,
+    completedTodosIds,
+    loadingTodosIds,
+    setLoadingTodosIds,
+    showError,
+    setShouldShowError,
+  } = useContext(AppContext);
+
+  const clearCompletedTodos = useCallback(async () => {
+    setShouldShowError(false);
+    setLoadingTodosIds(prevIds => [...prevIds, ...completedTodosIds]);
+
+    try {
+      await Promise.all(completedTodosIds.map(id => deleteTodo(id)));
+
+      setAllTodos(activeTodos);
+    } catch {
+      showError('Unable to delete all completed todos');
+    } finally {
+      setLoadingTodosIds([0]);
+    }
+  }, [loadingTodosIds, allTodos]);
```

Comment:
The useCallback dependencies array should include all used variables and functions from the component scope: [completedTodosIds, activeTodos, deleteTodo, showError, setAllTodos, setLoadingTodosIds, setShouldShowError]